### PR TITLE
Implement analytics :partying_face:

### PR DESCRIPTION
**Pull request by PatchMixolydic**

### CODE REVIEW

File: parsing_1.py [hunk 1]
```diff
@@ -39,4 +36,24 @@ def write_csv(counter):
         file_contents = r.content.decode().replace("\r\n", "\n")
         log.write(file_contents)
 
-write_csv(Counter(parse(file_contents)))
+# Extract the dates from every valid log entry
+date_strs = re.findall(APACHE_LOG_REGEX, log, re.MULTILINE)
+dates = [datetime.strptime(date, APACHE_LOG_DATE_FORMAT) for date in dates]
```
Comment (PatchMixolydic) [inline parsing_1.py:41]: This applies `datetime.strptime` to every date in `date_strs` and creates a new list with the results.

File: parsing_1.py [hunk 2]
```diff
@@ -39,4 +36,24 @@ def write_csv(counter):
...
+
+# This will allow us to binary search for the first date from six months ago
+dates.sort()
+last_date = dates[-1]
```
Comment (PatchMixolydic) [inline parsing_1.py:45]: `x[-1]` gets the last element in `x`

**Review by AnhPhantom — APPROVED** [review]
Great Job Ruby!!!

**Comment by PatchMixolydic** [comment]
Thank you!!

**Pull request merged**